Add Binomial-Normal BMA Meta-Analysis

FBartos opened the pull request:
also now:
fixes: https://github.com/jasp-stats/INTERNAL-jasp/issues/2129
fixes: https://github.com/jasp-stats/INTERNAL-jasp/issues/1201

juliuspfadt commented on R/bayesianmetaanalysisbinomial.R:
```diff
@@ -0,0 +1,457 @@
+  .robmaSummaryTable(jaspResults, options, type = "BiBMA")
+  # models overview
+  if (options[["inferenceModelsOverview"]])
+    .robmaModelsOvervievTable(jaspResults, options, type = "BiBMA")
```
```suggestion
    .robmaModelsOverviewTable(jaspResults, options, type = "BiBMA")
```

FBartos replied:
lol thanks!

juliuspfadt commented on R/bayesianmetaanalysisbinomial.R:
```diff
@@ -0,0 +1,457 @@
+  # create container for each of the parameters
+  for (parameter in c("effect", "heterogeneity")) {
+
+    if (!is.null(priorPlots[[parameter]]))
```
I would prefer that if you work with if else and one of them needs brackets that the other also gets brackets.

juliuspfadt commented:
I am not at all done, some preliminary things maybe: 

- I get an empty plot with this file. I have no idea if what I was doing made sense, the only data set I could find was the metaAnalsyis stuff form the data library
[emptyplot.jasp.zip](https://github.com/jasp-stats/jaspMetaAnalysis/files/14195847/emptyplot.jasp.zip)
- I do not like the name of the analysis in JASP. Why not call it "Binomial Meta-Analysis"? The "Binomial" in brackets just makes me think what should be in brackets for the Meta Analysis above it. But maybe that is the actual issue. If one if Binomial what is the other?

juliuspfadt commented on inst/qml/qml_components/RobustBayesianMetaAnalysisAdvanced.qml:
```diff
@@ -0,0 +1,244 @@
+				checked:			true
+				childrenOnSameRow:	true
+
+				DoubleField
```
I wonder if it makes sense to have this depend on the number of samples?

juliuspfadt commented on inst/qml/qml_components/RobustBayesianMetaAnalysisAdvanced.qml:
```diff
@@ -0,0 +1,244 @@
+				checked:			true
+				childrenOnSameRow:	true
+
+				DoubleField
+				{
+					name:			"advancedAutofitEssTarget"
+					defaultValue:	500
+					min:			1
+					inclusive:		JASP.None
+				}
```
I wonder if it makes sense to have this depend on the number of samples?

FBartos replied:
yeah, that would make sense, but I believe we don't do it at other places and I wouldn't like to have the autofit options change automatically once I adjust the rest of MCMC settings

juliuspfadt commented on inst/qml/qml_components/RobustBayesianMetaAnalysisAdvanced.qml:
```diff
@@ -0,0 +1,244 @@
+							min:			0
+						}
+
+						DropDown
+						{
+							name:			"advancedAutofitMaximumFittingTimeTargetUnit"
+							values:
+							[
+								{ label: qsTr("hours"),				value: "hours"},
+								{ label: qsTr("minutes"),			value: "mins"},
+								{ label: qsTr("seconds"),			value: "secs"}
+							]
+						}
```
I am curious, how do you measure the time?

FBartos replied:
`runjags` R package does that

juliuspfadt commented on inst/qml/qml_components/RobustBayesianMetaAnalysisDiagnostics.qml:
```diff
@@ -0,0 +1,114 @@
+	{
+		Group
+		{
+			title:			qsTr("Type")
```
So are all these plots? Is it the plot type of the Effect and Heterogeneity plots? If that is the case it should be better visible.

FBartos replied:
I'm not sure what you mean actually

juliuspfadt reviewed (changes requested):
Very nice! 
See the comments for potential improvements

FBartos commented on R/bayesianmetaanalysisbinomial.R:
```diff
@@ -0,0 +1,457 @@
+
+  # error handling
+  if (jaspBase::isTryError(fit))
+    .quitAnalysis(fit)
```
not really--this should never crash, but if it does, it at least gives users some package generated message to return

FBartos commented:
I fixed everything but one unit test--it works well when running interactively but running it via testAnalysis always creates slightly different results. No idea what's the issue though (I updated all packages etc).

FBartos commented:
the convergence check function does not refit the model, but the `update()` function in RoBMA does not update the convergence assessment properly (https://github.com/FBartos/RoBMA/issues/34)

I would just merge the PR and I will fix the package side bug separately.

juliuspfadt reviewed (approved):
sorry for taking so long
